zebrad/src/components/mempool.rs

```diff
@@ -322,3 +333,22 @@ impl Service<Request> for Mempool {
         }
     }
 }
+
+/// Remove transactions from the mempool if they have not been mined after a specified height.
+///
+/// https://zips.z.cash/zip-0203#specification
+fn remove_expired_transactions(
+    storage: &mut storage::Storage,
+    tip_height: zebra_chain::block::Height,
+) {
+    let ids = storage.tx_ids().iter().copied().collect();
+    let transactions = storage.transactions(ids);
+
+    for t in transactions {
+        if let Some(expiry_height) = t.transaction.expiry_height() {
+            if tip_height > expiry_height {
```

Comment:
I _think_ this should be `>=`

> If the expiry is at block height  N, then the transaction must be included in block N or earlier. Block N+1 will be too late, and the transaction will be removed from the mempool.

If `expiry_height` is N, then the transaction must be included in block N or earlier. But if `tip_height` is also N then the tx can't be included in it since it was just mined.

```suggestion
            if tip_height >= expiry_height {
```

Reply:
Thank you for that one, i changed this functionality at https://github.com/ZcashFoundation/zebra/pull/2774/commits/f7d8228783353e95cb49802e2327074cfc2bd8cb